model/plans.go

```diff
@@ -0,0 +1,224 @@
+// Copyright 2023 Northern.tech AS
+//
+//	Licensed under the Apache License, Version 2.0 (the "License");
+//	you may not use this file except in compliance with the License.
+//	You may obtain a copy of the License at
+//
+//	    http://www.apache.org/licenses/LICENSE-2.0
+//
+//	Unless required by applicable law or agreed to in writing, software
+//	distributed under the License is distributed on an "AS IS" BASIS,
+//	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//	See the License for the specific language governing permissions and
+//	limitations under the License.
+package model
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"regexp"
+	"sort"
+
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v3"
+)
+
+var (
+	validPlanName  = regexp.MustCompile(`^[a-zA-Z0-9_\\-]*$`)
+	PlanList       = []Plan{}
+	PlanNameToPlan = map[string]*Plan{}
+)
+
+type Features struct {
+	// RBAC
+	RBAC bool `json:"rbac" bson:"rbac"`
+
+	// audit logs
+	AuditLogs bool `json:"audit_logs" bson:"audit_logs"`
+
+	// dynamic groups
+	DynamicGroups bool `json:"dynamic_groups" bson:"dynamic_groups"`
+
+	// remote terminal
+	Terminal bool `json:"terminal" bson:"terminal"`
+
+	// file transfer
+	FileTransfer bool `json:"file_transfer" bson:"file_transfer"`
+
+	// configuration
+	Configuration bool `json:"configuration" bson:"configuration"`
+
+	// monitoring
+	Monitoring bool `json:"monitoring" bson:"monitoring"`
+}
+
+type PlanLimits struct {
+	// maximum number of devices
+	Devices int `json:"devices" bson:"devices"`
+
+	// maximum number of users
+	Users int `json:"users" bson:"users"`
+
+	// audit logs history in days
+	AuditLogsDays int `json:"audit_logs_days" bson:"audit_logs_days"`
+}
+
+type Plan struct {
+	// unique name used as identifier
+	Name string `json:"name" bson:"_id"`
+
+	// product
+	Product string `json:"product" bson:"product"`
+
+	// name to display
+	DisplayName string `json:"display_name" bson:"display_name"`
+
+	// feature set
+	Features *Features `json:"features" bson:"features"`
+
+	// limits
+	Limits *PlanLimits `json:"limits,omitempty" bson:"limits,omitempty"`
```

Comment:
As discussed, maybe we should keep the Limits only in the "PlanBinding" object.
```suggestion
```

Reply:
done

Reply:
Sorry, what did you do about the limits? Maybe I'm missing something :thinking:

Reply:
right; I forgot to include the last change and rebase... sorry; fixed now